File: Geometri CSharp/Geometri/Shape.cs

public abstract class Shape
{
    protected double a, b;

    public double A
    {
        get { return a; }
        set { a = value; }
    }
    public double B
    {
        get { return b; }
        set { b = value; }
    }


    protected Shape(double a, double b)
    {
        A = a;
        B = b;
    }

    public virtual double GetCircumference()
    {
        return (a * 2) + (b * 2);
    }
    public virtual double GetArea()
    {
        return a * b;
    }
}

File: Geometri C++/Shape.cpp
#include "Shape.h"

Shape::Shape(double a, double b) : a(a), b(b)
{}

double Shape::GetA()
{
    return a;
}
double Shape::GetB()
{
    return b;
}

void Shape::SetA(double a)
{
    this->a = a;
}
void Shape::SetB(double b)
{
    this->b = b;
}

double Shape::GetCircumference()
{
    return (a * 2) + (b * 2);
}
double Shape::GetArea()
{
    return a * b;
}
